### ISSUE TITLE
Proposal: New development and release cycle

### ISSUE THREAD

**Issue by lunny** [go-gitea/gitea#53]
I proposal a new development cycle.

# Objection

It should include stable versions, testing version, unstable version.

* Stable versions
Branches v0.1...v0.2 and tags v0.1.x ... v0.2.x.
** When release a new version like `v0.1`, it will create a branch named `v0.1` and a tag named `v0.1.0`. All bugs fixed should be pull request to `v0.1`. After some bug fixed, it could make a new tag named `v0.1.1` as v0.1's second release.
** When release a new version like `v0.2`, master should merge `v0.1` for bug fixed.

* Testing version
Master branch is a testing branch and all tested pull requests should be merged to master.

* Develop version
Develop branch & all forked branch

# Migration

From current development cycle to the new one, we only need create new branch at next release.

**Comment by tboerger** [comment]
Makes sense! Additional to that i would suggest some workflow around https://github.com/progrium/gh-release to upload built binaries to github.

**Comment by linquize** [comment]
So people should always open pull requests to `develop` branch?

**Comment by lunny** [comment]
No. If you pull request a bug fixed to a stable version, you just need pull request to the stable version branch. The different between the old and the new cycle is master is stable on old cycle and master is testing and not stable on new cycle.

**Comment by lunny** [comment]
The stable version will from master to v0.1 ... v0.x. @linquize

**Comment by makhov** [comment]
Hi from airport :)
What do you think about https://github.com/nvie/gitflow ?

**Comment by lunny** [comment]
It similarly likes as my proposal. @makhov

**Comment by tboerger** [comment]
> So people should always open pull requests to develop branch?

For new features? Of course! Bugfixes can be requested against ```develop``` as well, but if it makes sense it needs to be backported to the stable branches too.

**Comment by tboerger** [comment]
> Hi from airport :)
> What do you think about https://github.com/nvie/gitflow ?

Have a good fly! The gitflow behavior can be done quite easy with pure git commands as well :)

**Comment by linquize** [comment]
git-flow's master branch is unlike what @lunny said.

**Comment by lunny** [comment]
Yes. It's not different at this point. @linquize

**Comment by tboerger** [comment]
I think we can close that as it's decided or not?

**Comment by lunny** [comment]
I think we can close it after v0.7 is released. @tboerger
